### PR TITLE
feat: Parse JSON fields when content-type is set. Fixes #278.

### LINKED ISSUE #278: Add support for JSON objects for the non-file fields
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Currently the plugin doesn't support complex JSON objects as non-file field values. It could be an interesting addition to the plugin.

### Motivation

A user opened an issue where he is trying to send a JSON object through and then expects that to be validated.
Link to the issue is here: https://github.com/fastify/help/issues/525#issue-989998191

### Example

Below there's the code for both the server and the client (the non-file field in question is the **metadata** field).

server:
```JavaScript
const util = require("util");
const { pipeline } = require("stream");
const pump = util.promisify(pipeline);
const fs = require("fs");

const fastify = require("fastify")({
  logger: true,
  ajv: {
    customOptions: {
      coerceTypes: false,
      removeAdditional: false,
    },
  },
});

(async () => {
  const onFile = async (part) => {
    await pump(part.file, fs.createWriteStream(part.filename));
  };

  fastify.register(require("fastify-multipart"), {
    attachFieldsToBody: true,
    onFile,
  });

  fastify.post(
    "/upload/files",
    {
      schema: {
        body: {
          type: "object",
          required: ["metadata"],
          properties: {
            metadata: {
              type: "object",
              required: ["amount", "kind", "color"],
              properties: {
                amount: {
                  type: "string",
                },
                kind: {
                  type: "string",
                },
                color: {
                  type: "object",
                  required: ["main"],
                  properties: {
                    main: {
                      type: "string",
                    },
                    secondary: {
                      type: "string",
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
    async function (req, reply) {
      reply.send("ok");
    }
  );

  fastify.listen(3000, (err, address) => {
    if (err) throw err;
    console.info(`listening on port ${address}`);
  });
})();
```
client:
```JavaScript
const axios = require("axios");
const FormData = require("form-data");
const fs = require("fs");

const payload = {
  amount: "amount",
  kind: "kind",
  color: { main: "yellow", secondary: "green" },
};

const data = new FormData();
data.append("metadata", JSON.stringify(payload));

const config = {
  method: "post",
  url: "http://localhost:3000/upload/files",
  headers: {
    ...data.getHeaders(),
  },
  data: data,
  maxContentLength: Infinity,
  maxBodyLength: Infinity,
};

axios(config)
  .then(function (response) {
    console.log(JSON.stringify(response.data));
  })
  .catch(function (error) {
    console.log(error);
  });
```

### PR DESCRIPTION
**Pull request by ShogunPanda**
Fixes #278.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -340,6 +340,25 @@ hello: {
...
 ```
 
+#### JSON non-file fields
+
+If a non file field sent has `Content-Type` header starting with `application/json`, it will be parsed using `JSON.parse`. 
+
+The schema for validation for JSON fields should look like this:
+
+```js
+hello: {
```
[the remark this answers is not in the document]
Comment (ShogunPanda) [reply README.md:350]: I think I misexplained. I know we can use `allOf` there. But if you then wan to add the spec for the value field, then you have to use the syntax I showed in the first snippet in the comment. If we expose the schema as object as well, we can use destructuring like I did in my second snippet without using JSON schema keywords.

File: test/multipart-json.test.js [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+    try {
+      await pump(form, req)
+    } catch (error) {
+      t.error(error, 'formData request pump: no err')
+    }
+  })
+})
+
+test('should be able to use JSON schema to validate request when value is a JSON', function (t) {
```
Comment (Eomm) [inline test/multipart-json.test.js:273]: Could we test the `400` status code case?

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Eomm) [reply README.md:350]: Oh, I get it now.

I would just document it without exposing the JSON to descontruct

File: test/multipart-json.test.js (hunk 2 above)
Comment (ShogunPanda) [reply test/multipart-json.test.js:273]: If I got what you meant, it should be there now.

**Review by Eomm — APPROVED** [review]

**Pull request merged**